Add resource specifications documentation (#1892)
ubccr/xdmod: docs/configuration.md

For cloud resources the timezone is not used and times are converted to
the local timezone that the server is in.

> The `resource_allocation_type` option indicated how this resource is allocated

Reviewer: ```suggestion
The `resource_allocation_type` option indicates how this resource is allocated
```

to users, such as by CPU, GPU or Node. By default, there are 4 possible values,

> `CPU`, `CPUNode`, `GPU`, `GPUNode`.

Reviewer 2: Can you add a sentence to explain what the options mean. e.g. CPUNode allocation seems a bit oxymoronic.

For cloud resources the timezone is not used and times are converted to
the local timezone that the server is in.

The `resource_allocation_type` option indicates how this resource is allocated
to users, such as by CPU, GPU or Node. By default, there are 4 possible values,

> `CPU`, `CPUNode`, `GPU`, `GPUNode`. `CPUNode` denotes a system that allocates a
> node of CPU's to users.

Reviewer: ```suggestion
`CPU`, `CPUNode`, `GPU`, and `GPUNode`. `CPUNode` denotes a resource that allocates
nodes of CPUs to users, whereas `CPU` denotes a resource that allocates individual CPUs
to users.
```

multiple entries are required for that resource to calculate an accurate
utilization metric.

Note that the `end_date` is necessary for resources with multiple entries and

> still active. The `end_date` should not be include for the last entry for a

Reviewer: ```suggestion
still active. The `end_date` should not be included for the last entry for a
```

resource in `resources.json` should reference a resource type from this file.
This file typically should not be changed.

### resource_allocation_types.json

> Defines how a resource is allocated to user.  Each resource in `resources.json`

Reviewer: ```suggestion
Defines how a resource is allocated to users.  Each resource in `resources.json`
```
